airflow/contrib/operators/jira_operator.py

```diff
@@ -0,0 +1,92 @@
+
+    :param jira_conn_id: reference to a pre-defined Jira Connection
+    :type jira_conn_id: str
+    :param jira_method: method name from Jira Python SDK to be called
+    :type jira_method: str
+    :param jira_method_args: required method parameters for the jira_method
+    :type jira_method_args: dict
+    :param result_processor: function to further process the response from Jira
+    :type result_processor: function
+    :param get_jira_resource_method: function or operator to get jira resource
+                                    on which the provided jira_method will be executed
+    :type get_jira_resource_method: function
+    """
+
+    template_fields = ("jira_method_args",)
+
+    @apply_defaults
+    def __init__(self,
+                 jira_conn_id='jira_default',
+                 jira_method=None,
+                 jira_method_args=None,
+                 result_processor=None,
+                 get_jira_resource_method=None,
+                 *args,
+                 **kwargs):
+        super(JiraOperator, self).__init__(*args, **kwargs)
+        self.jira_conn_id = jira_conn_id
+        self.method_name = jira_method
+        self.jira_method_args = jira_method_args
+        self.result_processor = result_processor
+        self.get_jira_resource_method = get_jira_resource_method
+
+    def execute(self, context):
+        try:
+            # Currently this JiraOperator is limited to single operation only,
+            # it can be extended in future to perform multi step operations
+            # ex: using pythonOperator or taking function to operate on
+
+            if self.get_jira_resource_method:
+                """
+                if get_jira_resource_method is provided, jira_method will be executed on
+                resource returned by executing the get_jira_resource_method.
+                This makes all the provided methods of JIRA package accessible and usable
+                directly on the JiraOperator without needing a wrapper operator.
+                ref: http://jira.readthedocs.io/en/latest/api.html
+                """
+                if isinstance(self.get_jira_resource_method, JiraOperator):
+                    resource = self.get_jira_resource_method.execute(**context)
+                else:
+                    resource = self.get_jira_resource_method(**context)
+            else:
+                """ Default method execution is on the top level jira client resource."""
+                hook = JiraHook(jira_conn_id=self.jira_conn_id)
+                resource = hook.client
+
+            """
+            Current Jira-Python SDK (1.0.7) has issue with pickling the jira response.
+            ex: self.xcom_push(context, key='operator_response', value=jira_response)
+            This could potentially throw error if jira_result is not picklable
+            """
+            jira_result = getattr(resource, self.method_name)(**self.jira_method_args)
```

Comment:
Perhaps you could trap the error and log it, because otherwise the `AirflowException` will be raised and the Operator will be marked as failed even if it successfully changed the JIRA resource.